crates/fhir-sdk/src/client/r5/mod.rs

```diff
@@ -271,6 +270,91 @@ impl Client<FhirR5> {
 		.try_filter_map(|resource| async move { Ok(R::try_from(resource).ok()) })
 	}
 
+	/// Search for FHIR resources of a given type with query parameters.
+	/// This simply ignores resources of the wrong type, e.g. an additional
+	/// OperationalOutcome.
+	/// Contrary to [search], this will return a [Page] instance,
+	/// allowing manual pagination.
+	///
+	/// # Examples
+	/// ```no_run
+	/// use fhir_model::r5::resources::Person;
+	/// use fhir_sdk::client::{Client, Page, PagedSearchMethod, RequestSettings, SearchParameters};
+	///
+	/// async fn do_search() {
+	///     let client = Client::builder()
+	///         .base_url("http://127.0.0.1:8100/fhir/".parse().unwrap())
+	///         .request_settings(RequestSettings::default())
+	///         .build()?;
+	///     let parameters = SearchParameters::empty();
+	///
+	///     let page: Page<Person> = client.search_paged(PagedSearchMethod::Parameters(parameters)).await?;
+	///
+	///     let person_resources: Vec<Person> = page.results;
+	///     match page.next_url {
+	///         Some(url) => {
+	///             let new_page: Page<Person> = client.search_paged(PagedSearchMethod::Url(url)).await?;
+	///         }
+	///         None => {
+	///             println!("No next page");
+	///         }
+	///     }
+	/// }
+	///
+	/// ```
+	///
+	pub async fn search_paged<R: NamedResource + TryFrom<Resource, Error = WrongResourceType>>(
+		&self,
+		query: PagedSearchMethod,
+	) -> Result<Page<R>, Error> {
+		let url = match query {
+			PagedSearchMethod::Parameters(parameters) => {
+				let mut url = self.url(&[R::TYPE.as_str()]);
+				url.query_pairs_mut().extend_pairs(parameters.into_queries()).finish();
+				url
+			}
+			PagedSearchMethod::Url(url) => url,
+		};
+
+		if url.origin() != self.0.base_url.origin() {
+			return Err(Error::DifferentOrigin(url.to_string()));
+		}
+
+		let bundle: Option<Bundle> = self.read_generic(url.clone()).await?;
+		match bundle {
```

Comment:
Basically, unrolls the logic `Paged` follows. I'd like this to be simpler, maybe extracted to helper functions, but didn't see such patterns in the crate, so left this way.